Edit tags for a single album

chelliah opened the pull request:
Allows users to edit tags for a single album, from an AlbumTitle or an AlbumCard component. 

https://user-images.githubusercontent.com/6528140/213080928-e5cf3185-3572-4196-abe2-f6b6682adf7d.mov

This PR doesn't implement batch updates for editing multiple albums' tags at once, but that would be great to add later on.

lund5000 reviewed (changes requested):
I like the subtle button and the modal interaction works well. Good integration with the TagEdit database entity, transactions, and error recovery. 👍

One thing that isn't obvious and I don't think I'd detailed in the task yet: an album can only have one Local tag and one Rotation tag. Perhaps instead of a single checkbox group we need two radio button groups, each with a "None" or "Neither" option?

The other thing I'm wondering is whether we want to add a Boolean prop so we can decide when to show the edit button or not in different views. I think it should definitely be on the album view and tag view (so it's easy to quickly update heavy and light rotation every week), but it feels kind of noisy to me in crates, the music home, and even in search. @bkalb, what do you think?

bkalb commented on client/src/components/music/TagList.vue:
```diff
@@ -18,21 +29,25 @@ const allowedTags = [
         "d-none": this.filteredTags.length === 0,
       };
     },
+    canEditTags() {
```
I think it ought to be limited to users with the role "reviewer"

bkalb commented on client/src/components/crates/AlbumItem.vue:
```diff
@@ -8,7 +8,7 @@
       <CrateAlbumSpans :album="element.album" />
     </div>
-    <TagList :tags="element.album.current_tags" />
+    <TagList :tags="element.album.current_tags" :album="element.album" />
```
I'm fine with it only being visible on album view and tag view. Further out, it would be super cool to be able to select multiple albums and change their tags all at once, For rotation, it's most helpful to grab all albums that were added in a particular week and remove them from rotation.

chelliah replied:
Cool! Yeah would love to attempt a future refactor of this that accepts multiple albums as a prop and can edit them in parallel.

chelliah commented:
Addressed comments on this pr, let me know what you think @lund5000. Only thing I was still a little unsure about is the permissions check on `canEditTags`. I added a check for user roles but let me know if there is a better way to do this.

lund5000 commented:
It all looks good to me, including the permission-checking (which I tested with some users that don't have the "reviewer" role). In the interest of merging this PR and keeping it clean, I'm going to open up a follow-on PR to only show the edit button in the album and tag views.
